### PR TITLE
Hsph hack

### PR DESCRIPTION
**Pull request by benrudolph**
@czue this is initial implementation. @sheelio would like to test out on staging with forms that are similar to hsph to ensure this does what we want.

buddy: @biyeun 

https://github.com/dimagi/touchforms/pull/157

### CODE REVIEW

**Comment by czue** [comment]
kicked build. do you want this merged?

**Comment by sheelio** [comment]
Is this on Staging now?  

Going to test forms that use a main case, a parent case and the user case.  That seem right Ben?

**Comment by czue** [comment]
think @benrudolph is offline today, but it doesn't look like it's on staging yet

**Comment by czue** [comment]
build is failing because the submodule ref doesn't exist (maybe force pushed?)

**Comment by benrudolph** [comment]
yea i don't understand why committing the submodule didn't work. i didn't do any force pushing. sorry @sheelio this is not deployed yet. i think once i resolve this weird submodule thing we can just merge. it's behind a flag so not too risky

**Comment by sheelio** [comment]
Okay great - once merged and deployed I'll test it.

**Comment by benrudolph** [comment]
I think i should've fixed the tests, we can wait to build runs and then merge

**Comment by czue** [comment]
@benrudolph https://github.com/dimagi/touchforms/pull/157 still has "don't merge" on it - is that a mistake?

**Comment by benrudolph** [comment]
yes, removed the label. thanks for checking

File: corehq/apps/cloudcare/views.py [hunk 1]
```diff
@@ -263,6 +263,18 @@ def get_cases(request, domain):
     ids_only = string_to_boolean(request.REQUEST.get("ids_only", "false"))
     case_id = request.REQUEST.get("case_id", "")
     footprint = string_to_boolean(request.REQUEST.get("footprint", "false"))
+
+    if toggles.HSPH_HACK.enabled(domain):
+        hsph_case_id = request.REQUEST.get('hsph_hack', None)
+        if hsph_case_id != 'None' and hsph_case_id:
```
Comment (czue) [inline corehq/apps/cloudcare/views.py]: the comparing against the string `'None'` is kind of special

Comment (benrudolph) [inline corehq/apps/cloudcare/views.py]: yea unfortunately the current query manager we use in touchforms doesn't properly encode the parameters. that's why we have a bunch of `string_to_boolean` type functions floating around. probably should fix that

**Pull request merged**